### PR TITLE
fix: premature end large body

### PR DESCRIPTION
**Pull request by ronag**
llhttp integer truncation on ARM platform.

Refs: https://github.com/nodejs/undici/issues/803
Refs: https://github.com/nodejs/undici/issues/429

### CODE REVIEW

**Review by dnlup — APPROVED** [review]
Awesome work, lgtm.

**Comment by dnlup** [comment]
I think there's just some `ts` linting that is failing

File: lib/client.js [hunk 1]
```diff
@@ -834,6 +844,12 @@ class Parser {
       }
     }
 
+    /* istanbul ignore next: should be handled by llhttp? */
+    if (request.method !== 'HEAD' && contentLength && bytesRead !== parseInt(contentLength, 10)) {
+      util.destroy(socket, new ResponseContentLengthMismatchError())
+      return -1
+    }
```
Comment (mcollina) [inline lib/client.js:851]: Why is this ignored by istanbul? Can't we reproduce?

**Review by mcollina — APPROVED** [review]
lgtm with a nit

File: lib/client.js (hunk 1 above)
Comment (ronag) [reply lib/client.js:851]: I think llhttp doesn't actually allow it, i.e. it shouldn't be possible.

Comment (ronag) [reply lib/client.js:851]: I will have to dig into it further. Just too tired right now. Only slept 6 hours in total this week 😞 .

Comment (dnlup) [reply lib/client.js:851]: Please take some rest 🙏🏻

**Pull request merged**